Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1493,9 +1498,15 @@ LayoutUnit RenderFlexibleBox::marginBoxAscentForChild(const RenderBox& child)
 {
     auto ascent = alignmentForChild(child) == ItemPosition::LastBaseline ? child.lastLineBaseline() : child.firstLineBaseline();
     auto direction = isHorizontalFlow() ? HorizontalLine : VerticalLine;
-    if (!ascent) 
-        return synthesizedBaseline(child, style(), direction, BorderBox) + flowAwareMarginBeforeForChild(child);
-    return *ascent + flowAwareMarginBeforeForChild(child);
+    auto childMarginBefore = flowAwareMarginBeforeForChild(child);
+    if (!ascent) {
+        if (childHasScrollableOverflowCrossAxis(child))
+            return std::clamp(synthesizedBaseline(child, style(), direction, BorderBox) + childMarginBefore, childMarginBefore, childMarginBefore + crossAxisExtentForChild(child));
```

Comment:
isn't this just
`std::clamp(synthesizedBaseline(child, style(), direction, BorderBox), 0, crossAxisExtentForChild(child)) + childMarginBefore;`
and then the question is whether synthesizedBaseline() could really be < 0 as if not, it's just a `std::min(synthesizedBaseline(child, style(), direction, BorderBox), crossAxisExtentForChild(child)) + childMarginBefore;`

Reply:
Yes it is, but I actually don't think we need to do anything special here now that I look at it again. Since the synthesized baseline is based off the border box of the flex item, I think it should always provide a value inside the visible portion of the box even it is a scroll container.

For the similar scenario below, I think we still need to keep the logic (and I applied the change there also), we still need to keep the logic since its based off the content of the flex item